feat: JWT 인증 기반 팔로우 기능 구현

dain391 opened the pull request:
## 이슈 번호
#39

## 작업 내용
- [x] JwtUtil을 활용해 HTTP 요청 헤더의 JWT에서 userId 추출
- [x] 컨트롤러에서 userId를 파라미터로 받지 않고 JWT 기반으로 인증 처리
- [x] 서비스 메서드명 및 반환 타입 수정 및 일관성 유지
- [x] 불필요한 중복 제거 및 예외 처리 개선
- 
## 스크린샷(선택)

dain391 committed:
feat: JWT 인증 기반 팔로우 기능 구현 및 API 정리

- JwtUtil로 토큰에서 userId 추출해 인증 적용
- 팔로우 생성 및 조회 API 컨트롤러 수정
- 서비스 메서드명, 반환 타입 정리 및 예외 처리 개선

resolves: #39

taeung515 reviewed (commented):
1. build.gradle, jwtutil, .yml 파일 수정 해당 파일들은 다인님께서 맡으신 feature 범위가 아니므로 이번 커밋에 포함되지 않도록 수정해야합니다.

2. FollowService 계층에서 JwtUtil 클래스를 의존하실 필요가 없습니다. 이유는 이미 인증된 사용자만 follow 관련 URL에 접근 가능하도록 구현되어 있고 사용자 정보는 HttpServletRequest에 저장되어 Controller부에서 해당 값을 사용 가능하기 때문입니다. JwtUtil 클래스는 /login 요청 처리 시에만 사용됩니다.

3. 저희 응답형태는 return new ResponseEntity<>(ApiResponse.success(created), HttpStatus.CREATED); 형태로 강의에서 배운대로 통일하기로 하였으나 조회같은경우 return ResponseEntity.ok(followers);로 되어있습니다 따라서 이 부분도 통일형태로 수정해야합니다

4. return followRepository.findById(id).orElseThrow(() -> new IllegalArgumentException("해당 팔로우 관계를 찾을 수 없습니다. ID: " + id)); 이 부분도 통일하기로 한 응답형태에 어긋나나 @RestControllerAdvice 사용이 익숙치않다면 이정도만 해주셔도 될것 같습니다.


git branch dev // 데브브랜치로 이동
git  pull origin dev // 데브 최신화
git switch -c <새-브랜치-이름> // 데브브랜치에서 새로운 feature 브랜치 생성
기존에 구현하셨던 것들 복붙하셔서 위에 내용 보고 수정하시면 될 것 같습니다.
첨부사진 ( 매개변수로 HttpServletRequest 말고
@RequestAttribute(USER_ID) Long id
를 받으시면 됩니다.

dain391 committed:
WIP: 진행 중인 팔로우 기능 작업

dain391 committed:
fix: FollowService에서 JwtUtil 의존성 제거 및 API 응답 형식 통일

- JwtUtil 클래스 의존성을 FollowService에서 제거하고, Controller에서 @RequestAttribute(USER_ID)로 사용자 ID를 받도록 변경
- Follow 관련 API 응답을 ApiResponse.success() 형태로 통일하여 일관성 유지
- build.gradle, JwtUtil, application.yml 등 담당하지 않은 파일 커밋 제외 처리

resolves: #39

taeung515 reviewed (changes requested):
1. git branch dev // 데브브랜치로 이동
git pull origin dev // 데브 최신화
git switch -c <새-브랜치-이름> // 데브브랜치에서 새로운 feature 브랜치 생성
follow 하위 폴더만 복붙하셔서 다시 올리는 방법,


2. git merge dev 해서 충돌 직접 해결하는 방법 둘중 하나를 선택해주세요

dain391 committed:
chore: application.yml 파일 Git 추적에서 제외하고 .gitignore에 추가

application.yml 파일이 불필요하게 PR에 포함되는 문제를 해결하기 위해
Git 추적 대상에서 제거하고 .gitignore에 등록하였습니다.

resolves: #39

taeung515 commented:
[링크클릭](https://zoom.us/rec/play/1eUEU_dEcrRvK4Th9Co1dwoTqzlsUKGR9xChLc5GRKCwBBF4jip-h1wKYVwoLc_k9fTYMWcZgJ_t7zPS.Z9-IzkuXi5jCBmRx?eagerLoadZvaPages=sidemenu.billing.plan_management&accessLevel=meeting&canPlayFromShare=true&from=share_recording_detail&startTime=1744001414000&componentName=rec-play&originRequestUrl=https%3A%2F%2Fzoom.us%2Frec%2Fshare%2F6CGNmd8ljspDBgwnk8nWvGA57y3VrmxFyia5mCs0UFPyQaZ6J45aMSln0_vgIb6-.QxHjBDbboI6afys0%3FstartTime%3D1744001414000)

01:40:00 부터 보시면 병합을 왜해야하는지, 병합을 어떻게 진행하는지 알 수 있을겁니다. 참고하셔서 충돌해결 부탁드립니다

dain391 committed:
Merge branch 'dev' into feature/readFollow

# Conflicts:
#	newsfeed/build.gradle
#	newsfeed/src/main/resources/application.yml
